addons/website/static/src/components/dialog/add_page_dialog.js

```diff
@@ -227,9 +231,27 @@ export class AddPageTemplatePreview extends Component {
                 holderEl.classList.add("o_ready");
             };
             adjustHeight();
+            if (this.props.isCustom) {
+                this.adaptCustom(wrapEl);
+            }
         });
     }
 
+    adaptCustom(wrapEl) {
+        for (const sectionEl of wrapEl.querySelectorAll("section:not(.o_snippet_desktop_invisible)")) {
+            const style = window.getComputedStyle(sectionEl);
+            if (!style.height || style.display === 'none') {
+                const messageEl = document.createElement("section");
+                messageEl.classList.add("o_new_page_snippet_preview", "container", "h2");
+                const divEl = document.createElement("div");
+                divEl.classList.add("alert", "alert-info");
+                divEl.textContent = _t("No preview for the %s block because it is dynamically rendered.", sectionEl.dataset.name);
+                messageEl.appendChild(divEl);
```

Comment:
Maybe you could create the placeholder in the XML file and simply render it here ? :thinking: It will be easier to see what this part does :slightly_smiling_face: 

Also, instead of displaying an alert, isn't it possible to display images instead ? Personally, I would rather see an illustration looking like the snippet that will be on the page, so I have a good idea of the page in general (even if it the dynamic snippet does not have the same template in reality); than an alert telling that it will be rendered dynamically (and therefore preventing to visualize the real page) :thinking:

Reply:
Replaced by rendered template.
For the preview part, this is related to the PR about previewing dynamic content is various locations - so it will not be solved inside this PR.